src/transaction.cpp

```diff
@@ -1568,6 +1568,9 @@ int wsrep::transaction::certify_fragment(
         {
             ret = 1;
             error = wsrep::e_append_fragment_error;
+            storage_service.rollback(wsrep::ws_handle(),
```

Comment:
Would it be better to rollback only if `append_fragment()` fails? If `start_transaction()` fails for some reason, there might be also be failures during `rollback()` because there is no active transaction.